uint32 linear_distance
uint32 angular_distance
